blogapp/views.py

```diff
@@ -28,20 +28,22 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
     def retrieve(self, request, pk=None, *args, **kwargs):
-        queryset = BlogPost.objects.all()
-        blogpost = get_object_or_404(queryset, pk=pk)
+
+        blogpost = get_object_or_404(BlogPost, pk=pk)
```

Comment:
You should actually use queryset but the queryset should be defined at the beginning of the class